tests/integ/modin/binary/test_timedelta.py

```diff
@@ -70,23 +166,66 @@ def test_timestamp_minus_timestamp(self, scalar, operation):
                     ],
                 ]
             ),
-            lambda df: getattr(df, operation)(scalar),
+            lambda df: getattr(df, operation)(timestamp_scalar),
         )
 
-
-class TestSeriesAndScalar:
     @pytest.mark.parametrize(
-        "scalar",
+        "operation",
         [
-            pd.NaT,
-            datetime.datetime(year=2024, month=8, day=14, hour=2, minute=32, second=42),
-            datetime.datetime(year=2023, month=3, day=14),
-            pd.Timestamp(year=2020, month=3, day=25),
+            "add",
+            "radd",
+            "sub",
         ],
     )
+    @sql_count_checker(query_count=1)
+    def test_timestamp_dataframe_plus_or_minus_timedelta_scalar(
+        self, timedelta_scalar, operation
+    ):
+        eval_snowpark_pandas_result(
+            *create_test_dfs(
+                [
+                    [datetime.datetime(year=2024, month=1, day=1), pd.NaT],
+                    [
+                        datetime.datetime(year=2023, month=1, day=1),
+                        datetime.datetime(year=2030, month=1, day=1),
+                    ],
+                ]
+            ),
+            lambda df: getattr(df, operation)(timedelta_scalar),
+        )
+
+    @sql_count_checker(query_count=0)
+    def test_timedelta_scalar_minus_timestamp_dataframe_negative(self):
+        eval_snowpark_pandas_result(
+            *create_test_dfs([datetime.datetime(year=2024, month=8, day=21)]),
+            lambda df: pd.Timedelta(1) - df,
+            expect_exception=True,
+            expect_exception_type=TypeError,
+        )
+
+    @sql_count_checker(query_count=1)
+    @pytest.mark.parametrize("operation", ["add", "radd"])
+    def test_timedelta_dataframe_plus_timestamp_scalar(
+        self, timedelta_dataframes_1, timestamp_scalar, operation
+    ):
+        eval_snowpark_pandas_result(
+            *timedelta_dataframes_1, lambda df: getattr(df, operation)(timestamp_scalar)
+        )
+
+    @pytest.mark.parametrize("operation", ["sub", "rsub", "add", "radd"])
+    @pytest.mark.xfail(strict=True, raises=NotImplementedError, reason="SNOW-1637101")
+    def test_timedelta_dataframe_plus_or_minus_timedelta_scalar(
+        self, timedelta_dataframes_1, timedelta_scalar, operation
+    ):
+        eval_snowpark_pandas_result(
+            *timedelta_dataframes_1, lambda df: getattr(df, operation)(timedelta_scalar)
+        )
+
+
+class TestSeriesAndScalar:
```

Comment:
Doesn't necessarily need to happen in this PR, but might be neater to have a Timedelta folder here and separate out these tests into several files rather than have classes. Mostly just so running each file is quick, not a big deal.

Reply:
Hm, maybe. We can see how this file evolves. Right now with `pytest -n 32` the whole file runs very quickly locally.